workerapi/server/server.go

```diff
@@ -105,32 +118,49 @@ func (h *handler) QueryWorker() (*worker.WorkerStatus, error) {
 func (h *handler) Run(cmd *worker.RunCommand) (*worker.RunStatus, error) {
 	defer h.stat.Latency("runLatency_ms").Time().Stop()
 	h.stat.Counter("runs").Inc(1)
-	log.Infof("Worker Running:%s", cmd)
+	log.Infof("Worker trying to run cmd: %v", cmd)
 
 	h.updateTimeLastRpc()
 	c := domain.ThriftRunCommandToDomain(cmd)
-	process, err := h.run.Run(c)
+	status, err := h.run.Run(c)
+	//Check if this is a dup retry for an already running command and if so get its status.
```

Comment:
Is this check adequate? What if argv is the same but snapshot id or env is different?